### PR TITLE
Add min/max time filter to blockscopy and change min duration default to 0

### PR DESCRIPTION
**Pull request by pracucci**
I'm running a migration between two Grafana Mimir clusters and using the `blockscopy` tool to copy blocks. I've done some changes to `blockscopy` to support me, which I would like to upstream in this PR:

- Add min/max time filter (optional)
- Change min duration filter from 24h to 0 (so that all blocks get copied by default)

### CODE REVIEW

**Review by jhesketh — COMMENTED** [review]
Change seems reasonable to me. Should we have test coverage for this?

(The linting error appears unrelated).

**Comment by pracucci** [comment]
> Should we have test coverage for this?

We don't have tests for this tool. Ideally yes, but for many tools we don't have tests.

File: cmd/blockscopy/main.go [hunk 1]
```diff
@@ -50,7 +52,9 @@ type config struct {
 func (c *config) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&c.sourceBucket, "source-bucket", "", "Source GCS bucket with blocks.")
 	f.StringVar(&c.destBucket, "destination-bucket", "", "Destination GCS bucket with blocks.")
-	f.DurationVar(&c.minBlockDuration, "min-block-duration", 24*time.Hour, "If non-zero, ignore blocks that cover block range smaller than this.")
+	f.DurationVar(&c.minBlockDuration, "min-block-duration", 0, "If non-zero, ignore blocks that cover block range smaller than this.")
+	f.Var(&c.minTime, "min-time", "If set, only blocks with MinTime >= this value are copied.")
```
Comment (56quarters) [inline cmd/blockscopy/main.go:56]: Can you add an example of the expected format for these?

**Review by 56quarters — APPROVED** [review]
LGTM

File: cmd/blockscopy/main.go (hunk 1 above)
Comment (pracucci) [reply cmd/blockscopy/main.go]: Good idea. Done in: https://github.com/grafana/cortex-tools/pull/261/commits/ce4619c3ea97fa7f5d135ef2a597cf902c73fdfc

**Comment by pracucci** [comment]
Lint fails because of the old go version required. I'm going to merge cause I will move this tool to Mimir repo.

**Pull request merged**